### PR TITLE
Add type annotations and distribute them (PEP561)

### PR DESCRIPTION
**Pull request by hukkinj1**
Closes https://github.com/trezor/python-mnemonic/issues/72

- Add type annotations
- Distribute type data
- Check types using mypy

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,11 +15,15 @@ install:
...
     - pip install "pip>=7.0" wheel
     - pip install tox-travis
-    - pip install "black;python_version=='3.6'"
 
 script:
     - python setup.py install
-    - if [ $TRAVIS_PYTHON_VERSION = 3.6]; then black --check .; fi
+    - |
+      if [[ $TRAVIS_PYTHON_VERSION == '3.6' ]]; then
+        pip install black mypy &&
```
Comment (matejcik) [inline .travis.yml:23]: please keep this in the `install` section

File: mnemonic/mnemonic.py [hunk 2]
```diff
@@ -72,19 +79,19 @@ def __init__(self, language):
...
+    def list_languages(cls) -> List[str]:
         return [
             f.split(".")[0]
             for f in os.listdir(cls._get_directory())
             if f.endswith(".txt")
         ]
 
     @staticmethod
-    def normalize_string(txt):
+    def normalize_string(txt: Union[bytes, str]) -> str:
```
Comment (matejcik) [inline mnemonic/mnemonic.py:94]: `AnyStr` instead of this union?

File: mnemonic/py.typed [hunk 3]
```diff
@@ -0,0 +1 @@
+# Marker file for PEP 561.
```
Comment (matejcik) [inline mnemonic/py.typed:1]: please add a newline at end

**Comment by hukkinj1** [comment]
:+1: Made the changes

File: mnemonic/mnemonic.py (hunk 2 above)
Comment (hukkinj1) [reply mnemonic/mnemonic.py]: This is something for another PR, and a breaking change, but we could also consider removing `bytes` support from `normalize_string` altogether. Internally we never pass `bytes` to this method.

Comment (matejcik) [reply mnemonic/mnemonic.py]: True, but it doesn't seem worth it. Variability here costs us next to nothing.

**Pull request merged**